tests/api/test_account_meta.py

```diff
@@ -0,0 +1,77 @@
+import graphene
+import json
+import pytest
+from tests.api.utils import get_graphql_content
+
+META_LABEL = "TEST_LABEL"
+META_CLIENT = "TEST_PLUGIN"
+
+KEY = "name"
+VALUE = "Bond"
+
+
+@pytest.fixture
+def customer_with_meta(customer_user):
+    customer_user.store_private_meta(META_LABEL, META_CLIENT, {KEY: VALUE})
+    customer_user.save()
+    return customer_user
+
+
+GET_PRIVATE_META_QUERY = """
+    query UserMeta($id: ID!) {
+        user(id: $id) {
+            email
+            privateMeta {
+                label
+                clients {
+                    name
+                    metadata {
+                        key
+                        value
+                    }
+                }
+            }
+        }
+    }
+"""
+
+
+def test_get_private_meta(
+    staff_api_client, permission_manage_users, customer_with_meta
+):
+    user_id = graphene.Node.to_global_id("User", customer_with_meta.id)
+    variables = {"id": user_id}
+    staff_api_client.user.user_permissions.add(permission_manage_users)
```

Comment:
We never test the permissions of the resolver. I would recommend to pass the permission to `post_graphql` so it can ensure the permission works.

Reply:
Whole _user_ query is possible only when client has permission to manage users, hard to test in isolation

Reply:
That's what `post_graphql(..., permission=[manage_user]` is for, it checks if without the proper permissions the endpoint is rejecting the request, then it does the request with proper permissions. Unless I miss-understood.

Reply:
That's *highly* convoluted and implicit way to test things!
I did align my test with existing mechanism but it should probably be changed to some other way of ensuring access safety in more explicit manner.